### PR TITLE
Automate Local Build of Core Reference Documentation

### PR DESCRIPTION
**Pull request by yashincontrol**
Added copyReferenceFilesToStaging.js in common/scripts and modified gather-docs.yaml to call it.

### CODE REVIEW

File: common/scripts/copyReferenceFilesToStaging.js
Comment (pmconne) [inline common/scripts/copyReferenceFilesToStaging.js:29]: You assume the root repo directory is called "itwinjs-core" (not true on my machine), and your `basePath` assumes this script won't be relocated?
Considered using `git rev-parse` instead?

**Review by ben-polinsky — APPROVED** [review]

File: common/scripts/copyReferenceFilesToStaging.js
Comment (pmconne) [reply common/scripts/copyReferenceFilesToStaging.js]: I'm curious - under what circumstances would you `git rev-parse --show-toplevel` produce multiple lines of output?

**Review by pmconne — APPROVED** [review]

File: common/scripts/copyReferenceFilesToStaging.js
Comment (yashincontrol) [reply common/scripts/copyReferenceFilesToStaging.js]: The output from `git rev-parse --show-toplevel` seems to be ending with `"\n"`. That was causing problems with the rest of the code so I got rid of that.

**Comment by ben-polinsky** [comment]
@calebmshafer @wgoehrig @ColinKerr need one more review please.

File: common/scripts/copyReferenceFilesToStaging.js
Comment (aruniverse) [inline common/scripts/copyReferenceFilesToStaging.js:24]: wouldnt it be better to not swallow the error and log it? if its missing the dep it just throw an error anyway

File: common/scripts/copyReferenceFilesToStaging.js
Comment (aruniverse) [inline common/scripts/copyReferenceFilesToStaging.js:33]: should some of these catchs be fatal instead of swallowing/logging the error?

File: common/scripts/copyReferenceFilesToStaging.js
Comment (aruniverse) [inline common/scripts/copyReferenceFilesToStaging.js:53]: dont need an extra var for this

File: common/scripts/copyReferenceFilesToStaging.js
Comment (ben-polinsky) [reply common/scripts/copyReferenceFilesToStaging.js:24]: We thought it might be more clear to the user. Could throw.

File: common/scripts/copyReferenceFilesToStaging.js
Comment (ben-polinsky) [reply common/scripts/copyReferenceFilesToStaging.js:33]: The `return` effectively exits the program. Again, could throw.

File: common/scripts/copyReferenceFilesToStaging.js
Comment (aruniverse) [reply common/scripts/copyReferenceFilesToStaging.js:24]: i defer to you guys, this might be too nit picky. feel free to resolve both comments

Comment (ben-polinsky) [reply common/scripts/copyReferenceFilesToStaging.js:24]: Nah definitely good to consider.

**Review by aruniverse — APPROVED** [review]

**Pull request merged**
